src/client/components/draft/rich_text/test/rich_text.test.tsx

```diff
@@ -0,0 +1,1941 @@
+    })
+
+    it("Calls props.onChange if html is changed", done => {
+      const editorContent = convertFromHTML({})("<p>A new piece of text.</p>")
+      const editorState = EditorState.createWithContent(editorContent)
+      const component = getWrapper()
+      const instance = component.instance() as RichText
+      instance.onChange(editorState)
+
+      expect(component.state().html).toBe("<p>A new piece of text.</p>")
+      // Wait for debounced onChange
+      setTimeout(() => {
+        expect(component.instance().props.onChange).toHaveBeenCalled()
+        done()
+      }, 250)
+    })
+
+    it("Does not call props.onChange if html is unchanged", done => {
+      props.html = "<p>A new piece of text.</p>"
+      const editorContent = convertFromHTML({})(props.html)
+      const editorState = EditorState.createWithContent(editorContent)
+      const component = getWrapper()
+      const instance = component.instance() as RichText
+      instance.setState = jest.fn()
+      component.update()
+      instance.onChange(editorState)
+
+      setTimeout(() => {
+        expect(component.instance().setState).toBeCalled()
+        expect(component.instance().props.onChange).not.toHaveBeenCalled()
+        done()
+      }, 250)
+    })
+  })
+
+  describe("#focus", () => {
+    beforeEach(() => {
+      window.getSelection = jest.fn().mockReturnValue({
+        isCollapsed: false,
+        getRangeAt,
+      })
+    })
+
+    it("Calls focus on the editor", () => {
+      const component = getWrapper()
+      const instance = component.instance() as RichText
+      instance.focus()
+      const { hasFocus } = component.state().editorState.getSelection()
+
+      expect(hasFocus).toBeTruthy()
+    })
+
+    it("Calls #checkSelection", () => {
+      const component = getWrapper()
+      const instance = component.instance() as RichText
+      instance.checkSelection = jest.fn()
+      component.update()
+      instance.focus()
+
+      expect(instance.checkSelection).toBeCalled()
+    })
+  })
+
+  it("#blur removes focus from the editor", () => {
+    const component = getWrapper()
+    const instance = component.instance() as RichText
+    const selectedState = applySelectionToEditorState(
+      component.state().editorState
+    )
+    instance.onChange(selectedState)
+    instance.blur()
+    const { hasFocus } = component.state().editorState.getSelection()
+
+    expect(hasFocus).toBeFalsy()
+  })
+
+  describe("#resetEditorState", () => {
+    it("Resets the editorState with current props.html", done => {
+      const component = getWrapper()
+      const instance = component.instance() as RichText
+      props.html = "<p>A piece of text</p>"
+      component.setProps(props)
+      instance.resetEditorState()
+
+      setTimeout(() => {
+        expect(component.state().html).toBe("<p>A piece of text</p>")
+        done()
+      }, 10)
+    })
+  })
+
+  xdescribe("#componentWillReceiveProps", () => {
+    it("does a thing", () => {
```

Comment:
It is unclear if this function is currently necessary, still debugging from the parent. Will address in a separate PR.